### PR TITLE
Get jigsaw tests passing on csmbundle branch

### PR DESCRIPTION
**Pull request by krlberry**
## Description
With this update all old-format jigsaw app tests pass and also the new-format jigsaw tests pass. 
One Bundle-class unit test is still failing:
* isis_unit_test_BundleSolutionInfo (Failed)

Notes:
* I left the corrections, weights, adjustedsigmas, and apriorisigmas in `BundleObservation` for now. I think eventually they can probably come out and remain only in the parent class, but I didn't get around to getting this working.
* I removed all the large blocks of code that were already commented out in the Bundle classes I updated. My feeling was that if we we want to ultimately keep these sections of commented-out code, we can add them back in when the re-designed code is a bit more mature.
* Needed to keep SolveSettings in `AbstractBundleObservation` for `BundleMeasure` to compile, but I ultimately think it only belongs in the child classes. 
* I left comments in `BundleObservationVector` about things that need to be changed for ISIS vs. CSM observations, but these changes are not implemented.

## Related Issue
#4410 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [ ] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/control/objs/BundleUtilities/AbstractBundleObservation.cpp [hunk 1]
```diff
@@ -345,6 +330,8 @@ QString AbstractBundleObservation::formatBundleOutputString(bool errorPropagatio
    */
   QString AbstractBundleObservation::bundleOutputCSV(bool errorPropagation) {
     // different in both
+    // TODO: either remove or update. 
+    return "Test";
```
Comment (krlberry) [inline isis/src/control/objs/BundleUtilities/AbstractBundleObservation.cpp]: Do you think it's at all realistic to have something working for this at the parent class level, or should I just pull these out of this class?

File: isis/src/control/objs/BundleUtilities/AbstractBundleObservation.cpp [hunk 2]
```diff
@@ -317,6 +300,8 @@ namespace Isis {
  */
 QString AbstractBundleObservation::formatBundleOutputString(bool errorPropagation, bool imageCSV) {
   // different for both
+  // TODO: either remove or update. 
+  return "Test";
```
Comment (krlberry) [inline isis/src/control/objs/BundleUtilities/AbstractBundleObservation.cpp]: Do you think it's at all realistic to have something working for this at the parent class level, or should I just pull these out of this class?

File: isis/src/control/objs/BundleUtilities/AbstractBundleObservation.cpp (hunk 1 above)
Comment (jessemapel) [reply isis/src/control/objs/BundleUtilities/AbstractBundleObservation.cpp]: We can just make these functions pure virtual in the parent

File: isis/src/control/objs/BundleUtilities/AbstractBundleObservation.cpp (hunk 2 above)
Comment (jessemapel) [reply isis/src/control/objs/BundleUtilities/AbstractBundleObservation.cpp]: Same as CSV output, we can just make these pure virtual

**Review by jessemapel — APPROVED** [review]

**Pull request merged**